packages/builder/src/components/design/settings/controls/GridColumnConfiguration/PrimaryColumnFieldSetting.svelte

```diff
@@ -0,0 +1,101 @@
+<script>
+  import EditComponentPopover from "../EditComponentPopover.svelte"
+  import { Icon } from "@budibase/bbui"
+  import { setContext } from "svelte"
+  import { writable } from "svelte/store"
+  import { FieldTypeToComponentMap } from "../FieldConfiguration/utils"
+  import { store } from "builderStore"
+
+  export let item
+  export let anchor
+
+  let draggableStore = writable({
+    selected: null,
+    actions: {
+      select: id => {
+        draggableStore.update(state => ({
+          ...state,
+          selected: id,
+        }))
+      },
+    },
+  })
+
+  setContext("draggable", draggableStore)
+
+  const parseSettings = settings => {
+    return settings
+      .filter(setting => setting.key !== "field")
+      .map(setting => {
+        return { ...setting, nested: true }
+      })
+  }
+
+  const getIcon = () => {
+    const component = `@budibase/standard-components/${
+      FieldTypeToComponentMap[item.columnType]
+    }`
+    return store.actions.components.getDefinition(component).icon
+  }
+</script>
+
+<div class="list-item-body">
+  <div class="list-item-left">
+    <EditComponentPopover
+      {anchor}
+      componentInstance={item}
+      {parseSettings}
+      on:change
+    >
+      <div slot="header" class="type-icon">
+        <Icon name={getIcon()} />
```

Comment:
This won't be reactive, so if the `item` prop changes we wouldn't show the new icon. This is probably fine in this instance because the component probably remounts, but it would be better practice to derive the `icon` in a new reactive statement, then use that here.

Reply:
Good catch